src/Model/HasAttributesTrait.php

```diff
@@ -235,4 +260,19 @@ protected function buildMutatorMethod(string $key, string $prefix): string
     {
         return $prefix.Str::studly($key).'DocumentAttribute';
     }
+
+    private function shouldCastDateTime(string $key): bool
+    {
+        return array_key_exists($key, $this->casts)
+            && in_array($this->casts[$key], BaseDateTimeCast::$validCasts)
+            && !$this->attributes[$key] instanceof DateTimeInterface;
+    }
+
+
+    private function shouldCastUTCDateTime(string $key, mixed $value): bool
+    {
+        return array_key_exists($key, $this->casts)
+            && in_array($this->casts[$key], BaseDateTimeCast::$validCasts)
+            && !$value instanceof UTCDateTime;
+    }
```

Comment:
Sugestão, isso desacopla um pouco mais o mecanismo de caster do cast de dateTime.

```suggestion
    private function getCaster(string $key)
    {
        $casterName = $this->casts[$key] ?? null;

        return match($casterName) {
            'datetime' => app(DateTimeCast::class),
            'immutable_datetime' => app(DateTimeImmutableCast::class),
            default => null,
        }
    }
```